Lib/pathlib.py

```diff
@@ -364,7 +363,7 @@ def group(self, path):
 #
 # Globbing helpers
 #
-
+@functools.lru_cache
```

Comment:
`functools` provides pure python fallback incase c optimized one is not present so no need to check for it.